// @antv/component/lib/annotation/text.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var group_component_1 = require("../abstract/group-component");
var matrix_1 = require("../util/matrix");
var theme_1 = require("../util/theme");
var TextAnnotation = /** @class */ (function (_super) {
    tslib_1.__extends(TextAnnotation, _super);
    function TextAnnotation() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
     * @protected
     * 默认的配置项
     * @returns {object} 默认的配置项
     */
    TextAnnotation.prototype.getDefaultCfg = function () {
        var cfg = _super.prototype.getDefaultCfg.call(this);
        return tslib_1.__assign(tslib_1.__assign({}, cfg), { name: 'annotation', type: 'text', locationType: 'point', x: 0, y: 0, content: '', rotate: null, style: {}, defaultCfg: {
                style: {
                    fill: theme_1.default.textColor,
                    fontSize: 12,
                    textAlign: 'center',
                    textBaseline: 'middle',
                    fontFamily: theme_1.default.fontFamily,
                },
            } });
    };
    // 复写 setLocation 方法，不需要重新创建 text
    TextAnnotation.prototype.setLocation = function (location) {
        this.set('x', location.x);
        this.set('y', location.y);
        this.resetLocation();
    };
    TextAnnotation.prototype.renderInner = function (group) {
        this.renderText(group);
    };
    TextAnnotation.prototype.renderText = function (group) {
        var _a = this.getLocation(), x = _a.x, y = _a.y;
        var content = this.get('content');
        var style = this.get('style');
        var text = this.addShape(group, {
            type: 'text',
            id: this.getElementId('text'),
            name: 'annotation-text',
            attrs: tslib_1.__assign({ x: x,
                y: y, text: content }, style),
        });
        this.applyRotate(text, x, y);
    };
    TextAnnotation.prototype.applyRotate = function (textShape, x, y) {
        var rotate = this.get('rotate');
        var matrix = null;
        if (rotate) {
            matrix = matrix_1.getMatrixByAngle({ x: x, y: y }, rotate);
        }
        textShape.attr('matrix', matrix);
    };
    TextAnnotation.prototype.resetLocation = function () {
        var textShape = this.getElementByLocalId('text');
        if (textShape) {
            var _a = this.getLocation(), x = _a.x, y = _a.y;
            textShape.attr({
                x: x,
                y: y,
            });
            this.applyRotate(textShape, x, y);
        }
    };
    return TextAnnotation;
}(group_component_1.default));
exports.default = TextAnnotation;
