// @antv/g2/lib/core.js
"use strict";
/* G2 的一个壳子，不包含 Geometry，由开发者自己定义和引入 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VERSION = '4.0.7';
// 核心基类导出
var chart_1 = require("./chart"); // Chart, View 类
exports.Chart = chart_1.Chart;
exports.View = chart_1.View;
exports.Event = chart_1.Event;
var base_1 = require("./chart/controller/base"); // G2 组件基类
exports.ComponentController = base_1.Controller;
var base_2 = require("./geometry/base"); // Geometry 基类
exports.Geometry = base_2.default;
var base_3 = require("./geometry/label/base"); // Geometry Label 基类
exports.GeometryLabel = base_3.default;
var interaction_1 = require("./interaction"); // Interaction 基类
exports.Interaction = interaction_1.Interaction;
var facet_1 = require("./facet"); // Facet 基类
exports.Facet = facet_1.Facet;
var base_4 = require("./interaction/action/base"); // Interaction Action 基类
exports.InteractionAction = base_4.default;
// 注册 ComponentController
var chart_2 = require("./chart");
exports.registerComponentController = chart_2.registerComponentController;
// 注册 Geometry
var chart_3 = require("./chart");
exports.registerGeometry = chart_3.registerGeometry;
// 注册 Geometry Shape
var base_5 = require("./geometry/shape/base");
exports.registerShape = base_5.registerShape;
exports.registerShapeFactory = base_5.registerShapeFactory;
exports.getShapeFactory = base_5.getShapeFactory;
// 注册 Geometry label 以及 Geometry Label 布局函数
var label_1 = require("./geometry/label");
exports.registerGeometryLabel = label_1.registerGeometryLabel;
exports.registerGeometryLabelLayout = label_1.registerGeometryLabelLayout;
exports.getGeometryLabel = label_1.getGeometryLabel;
exports.getGeometryLabelLayout = label_1.getGeometryLabelLayout;
// 注册 interaction
var interaction_2 = require("./interaction");
exports.getInteraction = interaction_2.getInteraction;
exports.registerInteraction = interaction_2.registerInteraction;
exports.registerAction = interaction_2.registerAction;
exports.getActionClass = interaction_2.getActionClass;
// 注册 facet
var facet_2 = require("./facet");
exports.getFacet = facet_2.getFacet;
exports.registerFacet = facet_2.registerFacet;
// 注册主题
var theme_1 = require("./theme");
exports.getTheme = theme_1.getTheme;
exports.registerTheme = theme_1.registerTheme;
// G engine 管理相关
var engine_1 = require("./engine");
exports.registerEngine = engine_1.registerEngine;
exports.getEngine = engine_1.getEngine;
// 注册动画函数
var animation_1 = require("./animate/animation");
exports.registerAnimation = animation_1.registerAnimation;
exports.getAnimation = animation_1.getAnimation;
var constant_1 = require("./constant");
exports.LAYER = constant_1.LAYER;
exports.DIRECTION = constant_1.DIRECTION;
